service/service.go

```diff
@@ -113,7 +113,9 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 		MetricsLevel:   cfg.Telemetry.Metrics.Level,
 		// Construct telemetry attributes from build info and config's resource attributes.
 		Resource: pcommonRes,
-		Status:   status.NewReporter(srv.host.notifyComponentStatusChange),
+		Status: status.NewReporter(srv.host.notifyComponentStatusChange, func(err error) {
+			srv.telemetry.Logger().Warn("Invalid transition", zap.Error(err))
```

Comment:
I think `debug` is probably the proper log level. Failed state transitions are expected behavior when using the `ReportComponentStatus` API.

Reply:
Sure thing. Is this approach valid overall?

Reply:
It's an approach worth considering. My only reservation is that logging about invalid state transitions might give the user the impression that something is wrong with status reporting, when the invalid transitions are largely expected. I do think we want to give some feedback if the API is used before start however.

Reply:
ok that's doable by using errors.Is, just need to make sure the error is exported. I'll check.